src/ServiceInsight/Framework/RxServiceControl.cs

```diff
@@ -0,0 +1,181 @@
+﻿namespace ServiceInsight.Framework
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net.Http;
+    using System.Reactive;
+    using System.Reactive.Linq;
+    using System.Reactive.Subjects;
+    using System.Threading.Tasks;
+    using Akavache;
+    using Anotar.Serilog;
+    using ExtensionMethods;
+    using Newtonsoft.Json.Linq;
+    using ReactiveUI;
+    using Serilog;
+
+    interface IRxServiceControl
+    {
+        Task Refresh();
+
+        Task<bool> AddServiceControl(string url);
+
+        Task<bool> RemoveServiceControl(string url);
+
+        Task ClearServiceControls();
+
+        IObservable<ServiceControlData> Endpoints();
+
+        IObservable<ServiceControlData> Messages();
+
+        IObservable<ServiceControlData> EndpointMessages(string endpointName);
+
+        void SetRefresh(TimeSpan interval);
+
+        void DisableRefresh();
+    }
+
+    class RxServiceControl : IRxServiceControl
+    {
+        static Serilog.ILogger anotarLogger = Log.ForContext<IRxServiceControl>();
+
+        public static IRxServiceControl Instance { get; set; }
```

Comment:
No it's not registered as a singleton in the container.

I feel that with singletons we should start to take them out of the container to simplify the constructors for certain objects. I'm not sure about this. We should probably have a good discussion about it.